test/unit/gateways/realex_test.rb

```diff
@@ -51,6 +51,33 @@ def setup
     @amount = 100
   end
 
+  def test_initialize_sets_refund_and_credit_hashes
+    refund_secret = 'refund'
+    rebate_secret = 'rebate'
+
+    gateway = RealexGateway.new(
+      login: @login,
+      password: @password,
+      rebate_secret: rebate_secret,
+      refund_secret: refund_secret
+    )
+
+    assert gateway.options[:refund_hash] == Digest::SHA1.hexdigest(rebate_secret)
+    assert gateway.options[:credit_hash] == Digest::SHA1.hexdigest(refund_secret)
+  end
+
+  def test_initialize_with_nil_refund_and_rebate_secrets
```

Comment:
Would it make sense to do one more test variation here where we also assert that the hash computations are omitted when the `:rebate_secret` and `:refund_secret` options are actually omitted entirely (what the original guard did do correctly)?

Reply:
Good point. Done.